### PR TITLE
fix: ignore SAML issues in GH API

### PR DESCRIPTION
**Pull request by SMillerDev**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
This should allow people with SAML login for organisations that are not related to Homebrew (like me) to use commands that need the GH API.

### CODE REVIEW

File: Library/Homebrew/utils/github/api.rb [hunk 1]
```diff
@@ -322,7 +322,15 @@ def self.open_graphql(query, variables: nil, scopes: [].freeze, raise_errors: tr
 
       if raise_errors
         if result["errors"].present?
-          raise Error, result["errors"].map { |e| "#{e["type"]}: #{e["message"]}" }.join("\n")
+          related_errors = result["errors"].reject do |e|
+            e.key?("extensions") && e["extensions"]["saml_failure"] == true
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/github/api.rb:326]:
```suggestion
            # comment explaining why we don't care about saml_failure
            e.dig("extensions", "saml_failure") == true
```

**Review by MikeMcQuaid — COMMENTED** [review]
Works for me! Just wants a wee syntax tweak (use `dig`) and a comment.

**Comment by Bo98** [comment]
Just to clarify:

* Is this for interacting with third party taps or does it happen with Homebrew taps (and if so: why?)
* What happens when this error occurs? We're skipping the error here but are we actually get the data we wanted? Normally there isn't any data returned for errors.

**Comment by SMillerDev** [comment]
> What happens when this error occurs? 

I get a list of errors telling me to allow my Homebrew token access to the SAML orgs. And the command fails.

**Comment by MikeMcQuaid** [comment]
> Right now it only checks PRs created from non-forks (i.e. only covers maintainers)

This was not intended, whoops. Should be checking the base repository owner of the PR instead.

> The REST API query also seems to only be invoked if the user has >= 100 PRs across all of GitHub.

The, perhaps incorrect, intent here was that if they have <100 open PRs: we can instead iterate over them rather than having to hit the REST API.

**Pull request closed without merging**